libr/core/visual.c

```diff
@@ -4262,26 +4262,38 @@ R_API int r_core_visual(RCore *core, const char *input) {
 			} else {
 				ch = r_cons_readchar ();
 			}
-#ifndef __WINDOWS__
-			if (IS_PRINTABLE (ch) || ch == '\t' || ch == '\n') {
-				tcflush (STDIN_FILENO, TCIFLUSH);
-			} else if (ch == 0x1b) {
-				char chrs[2];
-				int chrs_read = 1;
-				chrs[0] = r_cons_readchar ();
-				if (chrs[0] == '[') {
-					chrs[1] = r_cons_readchar ();
-					chrs_read++;
-					if (chrs[1] >= 'A' && chrs[1] <= 'D') { // arrow keys
-						tcflush (STDIN_FILENO, TCIFLUSH);
-						// Following seems to fix an issue where scrolling slows
-						// down to a crawl after some time mashing the up and down
-						// arrow keys
-						r_cons_set_raw (false);
-						r_cons_set_raw (true);
+#ifdef __WINDOWS__
+			if (I->vtmode == 2 && !I->term_xterm) {
+#endif
+				if (IS_PRINTABLE (ch) || ch == '\t' || ch == '\n') {
+#ifdef __WINDOWS__
```

Comment:
Not a big fan of #ifdef all around... In the kernel they usually try to ifdef entire functions, so it would become something like:
```C
#ifdef __WINDOWS__
static void flush_stdin(void) {
   while (r_cons_readchar_timeout (1) != -1) ;
}
#else
static void flush_stdin(void) {
   tcflush (STDIN_FILENO, TCIFLUSH);
}
#endif
```

then you could just use that function both here and few lines below.

Reply:
Ok 43cbde42cb1544ac872bf3e59ae442b0676c1e7b